SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -201,48 +229,77 @@ void USpatialNetDriver::InitiateConnectionToSpatialOS(const FURL& URL)
 
 	if (!bPersistSpatialConnection)
 	{
-		// Destroy the old connection
 		GameInstance->DestroySpatialWorkerConnection();
-
-		// Create a new SpatialWorkerConnection in the SpatialGameInstance.
 		GameInstance->CreateNewSpatialWorkerConnection();
 	}
 
 	Connection = GameInstance->GetSpatialWorkerConnection();
 
-	if (URL.HasOption(TEXT("locator")))
+	bool bShouldLoadFromURL = true;
+	bool bUseReceptionist = true;
+
+	// If this is the first connection try using the command line arguments to setup the config objects.
+	// If arguments can not be found we will use the regular flow of loading from the input URL.
+	if (!GameInstance->bFirstConnectionToSpatialOSAttempted || URL.Host == SpatialConstants::RECONNECT_USING_ARGUMENTS)
 	{
-		// Obtain PIT and LT.
-		Connection->LocatorConfig.PlayerIdentityToken = URL.GetOption(TEXT("playeridentity="), TEXT(""));
-		Connection->LocatorConfig.LoginToken = URL.GetOption(TEXT("login="), TEXT(""));
-		Connection->LocatorConfig.UseExternalIp = true;
-		Connection->LocatorConfig.WorkerType = GameInstance->GetSpatialWorkerType().ToString();
+		FString CommandLineLocatorHost;
+		FParse::Value(FCommandLine::Get(), TEXT("locatorHost"), CommandLineLocatorHost);
+		if(!CommandLineLocatorHost.IsEmpty())
+		{
+			bShouldLoadFromURL = !Connection->LocatorConfig.TryLoadCommandLineArgs();
+			bUseReceptionist = false;
+		}
+		else
+		{
+			bShouldLoadFromURL = !Connection->ReceptionistConfig.TryLoadCommandLineArgs();
+			bUseReceptionist = true;
+		}
+
+		GameInstance->bFirstConnectionToSpatialOSAttempted = true;
```

Comment:
We should make this a private of the SpatialGameInstance and then have a method which can only set it to true.

Reply:
Done